docs/getting-started/webassembly-wasi.md

```diff
@@ -0,0 +1,340 @@
+import nodeResolve from "@rollup/plugin-node-resolve";
+
+export default {
+  input: "component.mts",
+  external: /wasi:.*/,
+  output: {
+    file: "dist/component.js",
+    format: "esm",
+  },
+  plugins: [
+    typescript({ noEmitOnError: true }),
+    nodeResolve(),
+  ],
+};
+```
+
+::: info
+Feel free to use any other bundlers that you're more comfortable with (`esbuild`, `rolldown`, etc)
+:::
+
+[jco]: https://github.com/bytecodealliance/jco
+[componentize-js]: https://github.com/bytecodealliance/componentize-js
+[rollup]: https://rollupjs.org/
+
+## 2. Retrieve WIT dependencies
+
+[`wasi:http`][wasi-http] is the standardized interface for dealing with HTTP requests (whether it's receiving them or sending them out),
+and since our component reuses that functionality, we have to declare it in the [WIT world][wit-world] our component exports:
+
+[wit-world]: https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#wit-worlds
+
+First, let's set up the component's WIT world:
+
+```wit
+/// wit/component.wit
+package example:hono;
+
+world component {
+    import wasi:cli/environment@0.2.3;
+
+    export wasi:http/incoming-handler@0.2.3;
+}
+```
+
+Put simply, the WIT file above means that we "make available" the functionality of "receiving" HTTP requests. While the above is
+relatively simple, it does depend on some upstream third party WIT interfaces (specifications on how requests are structured, etc).
+
+To pull those third party (Bytecode Alliance maintained) WIT interaces, one tool we can use is [`wkg`][wkg]:
+
+```sh
+wkg wit fetch
+```
+
+Once `wkg` has finished running, you should find your `wit` folder populated with a new `deps` folder alongside `component.wit`:
+
+```
+wit
+├── component.wit
+└── deps
+    ├── wasi-cli-0.2.3
+    │   └── package.wit
+    ├── wasi-clocks-0.2.3
+    │   └── package.wit
+    ├── wasi-http-0.2.3
+    │   └── package.wit
+    ├── wasi-io-0.2.3
+    │   └── package.wit
+    └── wasi-random-0.2.3
+        └── package.wit
+```
+
+[wkg]: https://github.com/bytecodealliance/wasm-pkg-tools
+[wasi-http]: https://github.com/WebAssembly/wasi-http
+
+## 3. Hello Wasm
+
+Let's fulfill our `component` world with a basic Hono application as a WebAssembly component:
+
+```ts
+// component.mts
+import { Hono } from 'hono'
+
+const app = new Hono();
+
+app.get('/hello', (c) => {
+  return c.json({
+    message: 'Hello WASM!',
+  })
+})
+
+/**
+ * Register the Hono application with the global fetch listener as supported by the underlying StarlingMonkey JS runtime.
+ *
+ * Since both Hono and StarlingMonkey are aligned Web Standards (WinterCG/WinterTC),
+ * this enables Hono to run smoothly in WASI-enabled (`wasi:http`) Webassembly environments.
+ *
+ * See: https://github.com/bytecodealliance/ComponentizeJS#using-starlingmonkeys-fetch-event
+ * See: https://hono.dev/docs/concepts/web-standard
+ * See: https://wintertc.org/
+ * See: https://github.com/WebAssembly/wasi-http
+ */
+app.fire();
```

Comment:
Now `app.fire()` is deprecated:

* https://github.com/honojs/hono/pull/4231
* https://github.com/honojs/hono/issues/3127

So, use `fire` function in `hono/service-worker`.

```ts
import { fire } from 'hono/service-worker'

// ...

fire(app)
```